packages/runtime-core/src/renderer.ts

```diff
@@ -2064,17 +2064,21 @@ function baseCreateRenderer(
   }
 
   const remove: RemoveFn = vnode => {
-    const { type, el, anchor, transition } = vnode
+    const { type, el, anchor, transition, children } = vnode
     if (type === Fragment) {
-      removeFragment(el!, anchor!)
+      const elementRoot = filterSingleRoot(children as VNodeArrayChildren)
```

Comment:
This doesn't look right. The whole fragment should be removed, not just the single root.

Reply:
I think the `removeFragment` function will remove the whole fragment?